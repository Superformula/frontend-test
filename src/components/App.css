:root {
  --white: #fff;
  --light-gray: #e6e6e6;
  --gray: #666666;
  --dark-gray: #333333;
  --black: #000;
  --blue: #002b56;
  --border: #c8c8c8;
  --green: #00e8a4;
  --red: #ff3548;
}

:root {
  --space-sm: 4px;
  --space-md: 8px;
  --space-lg: 16px;
  --space-xl: 24px;
  --space-2xl: 32px;
}

* {
  font-family: HelveticaNeue;
}
