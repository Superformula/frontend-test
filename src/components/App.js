import React from 'react'
import Router from './Router'

function App () {
  return (
    <div id='frontend-test'>
      <Router />
    </div>
  )
}

export default App
