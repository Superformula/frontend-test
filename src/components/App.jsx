import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Header from './Header';
import Filters from './Filters';
import RestaurantsList from './RestaurantsList';
import { fetchRestaurants, fetchCategories } from '../store/actions';
import Loader from './Loader';

const App = ({ restaurantsLoading, fetchRestaurants, fetchCategories, queryOffset }) => {
    useEffect(() => {
        // fetchCategories();
        fetchRestaurants(queryOffset);
    }, []);
    return (
        <div>
            <Loader loading={restaurantsLoading} />
            <Header />
            <div className="divider" />
            <Filters />
            <div className="divider" />
            <RestaurantsList />
        </div>
    );
};

App.propTypes = {
    restaurantsLoading: PropTypes.bool,
    queryOffset: PropTypes.number,
    fetchRestaurants: PropTypes.func,
    fetchCategories: PropTypes.func
};

const mapStateToProps = state => ({
    restaurantsLoading: state.restaurantsLoading,
    queryOffset: state.queryOffset
});
export default connect(
    mapStateToProps,
    {
        fetchCategories,
        fetchRestaurants
    }
)(App);
