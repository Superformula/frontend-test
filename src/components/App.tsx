import * as React from 'react'
import { Fragment, useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { useAsyncEffect } from 'use-async-effect'
import FilterRestaurants from './FilterRestaurants';
import RestaurantList from './RestaurantList';
import { updateRestaurants } from '../redux/actions/actions'


const App: React.FC = (): JSX.Element => {
  // Redux hook that allows actions to be dispatched
  const dispatch = useDispatch();

  // Special react hook, default useEffect has issues with async functions
  useAsyncEffect(async () => { // Executes on initial render. Replaces ComponentDidMount
    // Api call to GraphQL endpoint
    // Format data to an array of Restaurant Classes
    // Set redux state to new restaurants from GraphQL endpoint
      // dispatch(updateRestaurants()) -- final array is passed into updateRestaurants
  }, [])

  // Redux hook, grabbing current restaurants on redux store
  const restaurants = useSelector((state: any) => state.restaurants) 

  return (
    <Fragment>
      <h1>Restaurants</h1>
      <p>Browse through a curated list of your favorite restaurants</p>
      <FilterRestaurants />
      <RestaurantList rList={restaurants} />
    </Fragment>
  )
}

export default App;

