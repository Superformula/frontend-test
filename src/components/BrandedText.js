import * as React from 'react';

const styles = {
  h1: {
    fontWeight: "300",
    fontSize: "54px",
    color: "#333333",
    fontFamily: "Arial, Helvetica, sans-serif",
    padding: '15px 0px',
    display: "block"
  },
  h2: {
    fontWeight: "300",
    fontSize: "34px",
    color: "#333333",
    fontFamily: "Arial, Helvetica, sans-serif",
    display: "block"
  },
  h3: {
    fontWeight: "300",
    fontSize: "24px",
    color: "#333333",
    fontFamily: "Arial, Helvetica, sans-serif",
    display: "block"
  },
  p: {
    color: "#666666",
    fontSize: "22px",
    fontWeight: "lighter",
    fontFamily: "Arial, Helvetica, sans-serif",
    lineHeight: "25px",
    display: "block"
  },
  span: {}
};

export default ({ type, children }) => (
    <span
        style={styles[type]}
    >
        {children}
    </span>
);
