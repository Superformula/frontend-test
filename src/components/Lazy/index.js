import React, { Suspense } from 'react';
import PropTypes from 'prop-types';

export const Lazy = ({ component: Component }) => (
  <Suspense fallback={'loading...'}>
    <Component />
  </Suspense>
);

Lazy.propTypes = {
  component: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
};

export default Lazy;
