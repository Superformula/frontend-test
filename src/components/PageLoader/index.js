import React from 'react';
import { Background } from './Background';
import { Loader } from './Loader';

export const PageLoader = () => (
  <Background>
    <Loader />
  </Background>
);
