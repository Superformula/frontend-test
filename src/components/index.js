export { default as Card } from './Card/Card';
export { default as CardSkeleton } from './Card/CardSkeleton';
export { default as StarRating } from './StarRating/StarRating';
export { default as Button } from './Button/Button';
export { default as Badge } from './Badge/Badge';

export { default as Headline } from './Typography/Headline';
export { default as Heading } from './Typography/Heading';
export { default as SubHeading } from './Typography/SubHeading';
export { default as Lead } from './Typography/Lead';
export { default as Small } from './Typography/Small';

export { default as Container } from './Grid/Container';
export { default as Row } from './Grid/Row';
export { default as GridItem } from './Grid/GridItem';

export { default as Label } from './Forms/Label/Label';
export { default as Checkbox } from './Forms/Checkbox/Checkbox';
export { default as SelectDropDown } from './Forms/SelectDropDown/SelectDropDown';

export { default as FilterNav } from './FilterNav/FilterNav';
export { default as HeroMini } from './HeroMini/HeroMini';
export { default as BusinessList } from './RestaurantList/RestaurantList';
