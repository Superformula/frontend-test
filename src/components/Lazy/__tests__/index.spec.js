import React from 'react';
import { shallow } from 'enzyme';
import { PageLoader } from 'components/PageLoader';
import Lazy from '../';

describe('Lazy', () => {
  it('should render without failing', () => {
    const Test = () => <div />;
    const wrapper = shallow(<Lazy component={Test} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should use the loader as a fallback', () => {
    const Test = () => <div />;
    const wrapper = shallow(<Lazy component={Test} />);
    const { fallback } = wrapper.find('Suspense').props();
    expect(fallback).toEqual(<PageLoader />);
  });
});
