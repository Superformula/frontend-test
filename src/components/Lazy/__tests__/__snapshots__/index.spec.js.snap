// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lazy should render without failing 1`] = `
<Suspense
  component={[Function]}
  fallback={<PageLoader />}
>
  <Test />
</Suspense>
`;
